lib/root_loader.rb

```diff
@@ -33,6 +35,17 @@ def load account, filename
 
         loader = Loader::Orchestrate.new policy_version
         loader.load
+
+        created_roles = loader.new_roles.select do |role|
+          %w(user host).member?(role.kind)
+        end.inject({}) do |memo, role|
+          credentials = Credentials[role: role] || Credentials.create(role: role)
+          memo[role.id] = { id: role.id, api_key: credentials.api_key }
+          memo
+        end
```

Comment:
I'd split this whole thing up into 2 statements, rather than chaining.

Also, see how the above has the awkward pattern 
```ruby
memo[blah] = stuff.....
memo
```
`each_with_object` is made for avoiding just this.

Reply:
So this is exactly what I meant.... buuuuuut

I don't get how it's working because `each_with_object` reverses the arguments relative to `reduce`, so that the memo is in the 2nd arg:

https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-each_with_object